client/src/main/com/sinch/sdk/domains/verification/StatusService.java

```diff
@@ -0,0 +1,49 @@
+package com.sinch.sdk.domains.verification;
+
+import com.sinch.sdk.domains.verification.models.Identity;
+import com.sinch.sdk.domains.verification.models.VerificationMethodType;
+import com.sinch.sdk.domains.verification.models.VerificationReport;
+
+/**
+ * Get the status of specific verification requests in the verification process. Returns the status
+ * of pending and completed verifications. You can retrieve the status of verification requests by
+ * using the ID of the request, the phone number of the user being verified, or a custom reference
+ * string.
+ *
+ * @see <a
+ *     href="https://developers.sinch.com/docs/verification/api-reference/verification/tag/Verification-status">https://developers.sinch.com/docs/verification/api-reference/verification/tag/Verification-status</a>
+ * @since 1.0
+ */
+public interface StatusService {
```

Comment:
The same way you are using the same name for `report` because the signatures are different, can we imagine having just one `get` (and define classes (or records) for `Id` and `Reference`)

Reply:
I'd like too; but it won't fit.

- Record usage: not for currently java 8 to be supported 
- Overriding: String for `id` vs `reference`: it will imply for casting onto calls with possibly end user "errors" when having to call:
  - status.get((myidclass casting) "a string id")
  - status.get((myreferenceclass casting) "a string  reference")

Reply:
huum.. let me check a little bit more ..
I do not want to much over-engineering around simple "String" but may be we can imagine something like fo Double, Float, ...: `static Id Id.valueOf('String" idValue)` but this will imply REST response classes have to be designed to use these types
to be continued

Reply:
@asein-sinch please review this implementation: https://github.com/sinch/sinch-sdk-java/pull/16/commits/0a472438085a8cbe129ec049df8b4d52fabf1ada

Reply:
A question I asked myself :-)
Because of I was mimic the Float/Double/... java classes with function like `intValue`, `floatValue`... I did not retains that way and kept dedicated function names (that could have been `idValue`/`referenceValue` according to java classes naming, but stay attached to `getters` usage from SDK)